### DISCUSSION TITLE
Filter list while typing?

### DISCUSSION THREAD

**Discussion by ghost** [chakra-ui/chakra-ui#2830] [Q&A] [answered]
Is there a way to use Chakra to create something LIKE a Select box, but more like a COMBO box, where the user can type and the list filters down to items that match the typed characters?

So, if I had a list like this:

1000
1001
1010
1100
2010
2011
2200
2210
2305

when I type "2", it only shows the items starting with 2, filtering out all the others.
after that, if i typed another 2 ("22), the list above would only have two items in it:

2200
2210

and then if I typed a 1 ("221"), there'd only be a single item in the list:

2210.

At any point, I could tap an item in the list to select it (replacing the typed value with the full value of the tapped item)

Any way to do that in Chakra?

**Comment by chr-ge** [comment, marked as answer]
This is something you would have to implement yourself or use another package like [react-select](https://github.com/JedWatson/react-select).

**Reply by ghost** [reply]
That's a well designed 3rd party component. Looks like it will work perfectly for my needs. Thanks!

**Reply by csandman** [reply]
If you want a version of react-select that looks more like a Chakra component, check out my wrapper for it!

https://github.com/csandman/chakra-react-select

![image](https://user-images.githubusercontent.com/9214195/149231131-0c033705-730b-42d6-9117-dcdbe556e3cc.png)
